testbed/tests/window/test_window.py

```diff
@@ -138,13 +143,191 @@ async def test_move_and_resize(main_window, main_window_probe, capsys):
+        if (
+            not main_window_probe.supports_fullscreen
+            and state == WindowState.FULLSCREEN
+        ):
+            pytest.xfail("This backend doesn't support fullscreen window state.")
+        if (
+            not main_window_probe.supports_presentation
+            and state == WindowState.PRESENTATION
+        ):
+            pytest.xfail("This backend doesn't support presentation window state.")
+
+        try:
+            # Set the window state:
+            main_window.state = state
+            await main_window_probe.wait_for_window(f"Secondary window is in {state}")
+            assert main_window.state == state
+
+            # Set the window state the same as current:
+            main_window.state = state
+            assert main_window.state == state
+
+        finally:
+            # Restore to NORMAL state.
+            main_window.state = WindowState.NORMAL
+            await main_window_probe.wait_for_window("Main window is not full screen")
+            assert main_window.state == WindowState.NORMAL
+
+    async def test_window_state_fullscreen(main_window, main_window_probe):
+        """The window can enter into fullscreen state."""
+        if not main_window_probe.supports_fullscreen:
+            pytest.xfail("This backend doesn't support fullscreen window state.")
+        try:
+            widget = toga.Box(style=Pack(flex=1))
+            widget_probe = get_probe(widget)
+            main_window.content = toga.Box(children=[widget])
+            await main_window_probe.wait_for_window(
+                "Test widget has been added to the window"
+            )
+            widget_initial_size = (widget_probe.width, widget_probe.height)
+
+            # Make main window full screen
+            main_window.state = WindowState.FULLSCREEN
+            await main_window_probe.wait_for_window(
+                "Main window is full screen", full_screen=True
+            )
+            assert main_window.state == WindowState.FULLSCREEN
+            # At least one of the dimensions should have increased.
+            assert (
+                widget_probe.width > widget_initial_size[0]
+                or widget_probe.height > widget_initial_size[1]
+            )
+        finally:
+            # Exit full screen
+            main_window.state = WindowState.NORMAL
+            await main_window_probe.wait_for_window(
+                "Main window is not full screen", full_screen=True
+            )
+            assert main_window.state == WindowState.NORMAL
+            # Both dimensions should be the same.
+            assert (
+                widget_probe.width == widget_initial_size[0]
+                and widget_probe.height == widget_initial_size[1]
+            )
+            main_window.content.clear()
+
+    async def test_window_state_presentation(main_window, main_window_probe):
+        """The window can enter into presentation state."""
```

Comment:
How is this any different to the "NORMAL->PRESENTATION" state change test?

Reply:
It has an extra check for ensuring that the widgets with `flex=1` increase one or both of their dimensions, when the app enters PRESENTATION mode.